ui/src/hooks/models.tsx

```diff
@@ -191,3 +190,40 @@ export function useDeleteModelVersion({
     },
   });
 }
+
+export interface DeleteModelMutationParams
+  extends Pick<ModelInterface, 'catalog_name' | 'schema_name' | 'name'> {}
+
+interface DeleteModelParams {
+  catalog: string;
+  schema: string;
+  model: string;
+}
+
+// Delete a model
+export function useDeleteModel({ catalog, schema, model }: DeleteModelParams) {
+  const queryClient = useQueryClient();
+
+  return useMutation<void, Error, DeleteModelMutationParams>({
+    mutationFn: async ({
+      catalog_name,
+      schema_name,
+      name,
+    }: DeleteModelMutationParams) => {
+      const fullName = [catalog_name, schema_name, name].join('.');
+      return apiClient
+        .delete(`/models/${fullName}`)
+        .then((response) => response.data)
+        .catch((e) => {
+          throw new Error(
```

Comment:
Will this show what the exception was?  Concretely I'm asking if we will tell them about a permission denied error if they try to delete something they cannot.

Reply:
Yes, it will show the error message from the server side, if there is one. Otherwise, it will show a generic "failed to delete model" message.